### DISCUSSION TITLE
PFSoC bare metal examples for the Video Kit

### DISCUSSION THREAD

**Discussion by rockoguido** [polarfire-soc/polarfire-soc-documentation#321] [Bare metal embedded software] [answered]
Hi..

Trying to support a customer using the Video Kit (MPFS250-VIDEO-KIT).    I am in process of trying to get one but don't have one yet.   Question is could a user use the Bare Metal Examples (I.e. mpfs-hal-ddr-demo) on the Video Kit and perhaps use the Board Support for the Icicle Kit or another board called out in the src/boards directory?  

Thx!
Dave

**Comment by hughbreslin** [comment, marked as answer]
The DDR demo specifically has build targets for the different kits available :) Just note it was actually called SEV kit in the demo by accident (Smart Embedded Vision instead of Video Kit and should be renamed in a future release):

![image](https://user-images.githubusercontent.com/44862881/235899276-4a3f8bc4-7a23-4c8f-8d3c-bb2b911b1156.png)

Once you have the peripherals enabled and the correct fabric set up (i.e anything routed through the fabric actually connected up) you shouldn't have any issue running the bare metal examples - you can just add a video kit board (or copy the video kit board from the DDR demo if you want the base configuration) and add it to your project with the include paths set up to point to it as a build configuration.

**Reply by rockoguido** [reply]
thanks, good to know the video kit support is there, just in stealth mode.    It's interesting that for the HSS the video kit support is more explicit but that didn't make it into the Bare Metal Examples

I do have a follow-on question.   it's not clear how you got the pulldowns that you show above, these look like a list of C/C++ Build Configurations and I do see a SEV-Kit-LIM-Debug configuration, just not the reference to LPDDR4 on SEV.   Perhaps the bare metal examples have been rev'ed lately?   I'll take a look

**Reply by hughbreslin** [reply]
Yes I am trying to see what we can do about the other examples, it isn’t a huge process to get them going to but having it out of the box would be ideal.

Re my build configs, tbh I’m not sure I could have an internal copy of the example (I thought it was the public one), I just imported the ddr demo from the bare metal examples repo and those were the builds bundled with the project. My version could be a couple of months old but the build configs should be bundled for the boards we have

**Reply by rockoguido** [reply]
ok had a chance to give it a try, just grabbed the bare metal examples repo from
 https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples.  From there just imported into a fresh workspace the mpfs-hal-ddr-demo, and I'm able to see all of those various build configurations.   very cool...

So, for my customer using the Video Kit it turns out the bare metals examples repo that I just list is out of date vs the GitHub page for the video kit at..
https://github.com/polarfire-soc/polarfire-soc-video-kit-reference-design
The issue was that the customer's DDR training was failing, so he just replaced the board specific xml and cfg files from the Bare Metal Repo with the files from the Video Kit page and problem solved.
However it highlights a file synchronization issue we have.
Thx!
Dave..   aka rockoguido

**Reply by nitindeshpande** [reply]
The renaming from SEV to Video kit is already done. It is scheduled for release this month.
